src/cartridge.rs

```diff
@@ -33,4 +36,14 @@ impl Cartridge {
         };
         Ok(cartridge)
     }
+
+    pub fn load_power_up(path: &str, cartridge: &mut Vec<u8>) -> Result<(), io::Error> {
+        let mut file = std::fs::File::open(path)?;
+        let mut buffer = vec![];
+        file.read_to_end(&mut buffer)?;
+        unsafe {
+            ptr::copy_nonoverlapping(buffer.as_ptr(), cartridge.as_mut_ptr(), buffer.len());
+        }
```

Comment:
This overflows if the rom given as argument is incorrect and has a len smaller than the startup sequence. Either make this function return an error or make this function unsafe to indicate the check must be made before calling it

Reply:
af902b94baafda73d52e36b05288db3f3b7a9a45

Reply:
3981d25a2007d101480fc0332a70957eddc27afa

Reply:
This was not solved

Reply:
d3393b2

Reply:
bf9ac14

Reply:
db270b735ca4671f258b1ac04433cf6861f97ff1